src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -526,6 +530,49 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     LOG_DEBUG("CompiledModel is being deserialized, skipping the full constructor flow...");
 }
 
+void ov::npuw::CompiledModel::identify_host_gather_property(const std::shared_ptr<ov::Model>& model) {
+    // Identify based on the config
+    try {
+        auto compiler_version = get_plugin()->get_core()->get_property("NPU", ov::intel_npu::compiler_version);
+        if (compiler_version >= ONEAPI_MAKE_VERSION(7, 21)) {
+            // Force quantized host gather
+            LOG_INFO("Forcing NPUW_HOST_GATHER_QUANT property.");
+            std::map<std::string, std::string> host_gather_cfg;
+            host_gather_cfg["NPUW_HOST_GATHER_QUANT"] = "YES";
+            host_gather_cfg["NPUW_HOST_GATHER"] = "NO";
+            m_cfg.update(host_gather_cfg);
+        } else {
+            if (m_cfg.get<::intel_npu::NPUW_HOST_GATHER_QUANT>()) {
+                NPUW_ASSERT(false && "NPU compiler doesn't support tail quantization! "
+                                     "Please either use NPUW_HOST_GATHER instead of NPUW_HOST_GATHER_QUANT "
+                                     "or get the latest driver.");
+            }
+        }  // compiler_version
+    } catch (...) {
+        // No device, keeping properties as is. Just do a sanity check
```

Comment:
hmm, I believe we queried available_devices before? I also remember it caused some issues with GPU resource handling in some of our benchmarks.

Reply:
@PatrikStepan @sbutnari can you guys recommend a gentler way to test if the NPU is present?

I believe we don't really check for the NPU device but for the NPU compiler as well (having an NPU-less system but with CIP is probably a legit case?)